apps/ui/src/components/SpaceTreasury.vue

```diff
@@ -105,6 +105,10 @@ const executionStrategy = computed(() => {
   };
 });
 
+const hasStakeableAssets = computed(() => {
+  return assets.value.some(asset => asset.contractAddress === ETH_CONTRACT && !isReadOnly);
```

Comment:
```suggestion
  return !isReadOnly && assets.value.some(asset => asset.contractAddress === ETH_CONTRACT);
```

We only need to check if it's read only once, and if it is stop `some` from executing entirely.